grails-app/controllers/org/pih/warehouse/invoice/InvoiceController.groovy

```diff
@@ -102,6 +111,37 @@ class InvoiceController {
         redirect(action: "create", params: [id: invoice.id])
     }
 
+    def generateSalesInvoiceFromShipment = { Invoice invoice ->
+        JSONObject jsonObject = new JSONObject()
+
+        Location currentLocation = Location.get(session.warehouse.id)
+        if (!currentLocation) {
+            throw new IllegalArgumentException("User must be logged into a location to create invoice")
+        }
+
+        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
+        Shipment shipment = stockMovement.shipment
+        orderService.createOrderFromShipment(shipment, currentLocation)
+
+        jsonObject.put("party", shipment?.destination?.organization)
```

Comment:
What's the point of using jsonObject here? Remove it and pass into the bindSalesInvoiceData the shipment and just assign appropriate properties.

Reply:
Right, it is not needed anymore.